File: sunflower-core/src/main/java/org/tdf/sunflower/net/MessageLogger.java
package org.tdf.sunflower.net;

import lombok.extern.slf4j.Slf4j;

@Slf4j(topic = "net")
class MessageLogger implements Plugin {
    @Override
    public void onMessage(ContextImpl context, PeerServerImpl server) {
        log.debug("receive {} message from {}:{}",
                context.message.getCode(), context.remote.getHost(), context.remote.getPort()
        );
        log.debug(context.message.toString());
    }

    @Override
    public void onStart(PeerServerImpl server) {

    }

    @Override
    public void onNewPeer(PeerImpl peer, PeerServerImpl server) {
        log.info("new peer join {}", peer);
    }

    @Override
    public void onDisconnect(PeerImpl peer, PeerServerImpl server) {
        log.info("peer disconnected {}", peer);
    }

    @Override
    public void onStop(PeerServerImpl server) {

    }
}

File: sunflower-core/src/main/java/org/tdf/sunflower/net/MessageLogger.kt
package org.tdf.sunflower.net

import org.slf4j.LoggerFactory

internal class MessageLogger : Plugin {
    override fun onMessage(context: ContextImpl, server: PeerServerImpl) {
        log.debug(
            "receive {} message from {}:{} udp = {} size = ${context.msg.serializedSize}",
            context.msg.code, context.remote.host, context.remote.port,
            context.udp
        )
        log.debug("data = {}", context.msg)
    }

    override fun onStart(server: PeerServerImpl) {}
    override fun onNewPeer(peer: PeerImpl, server: PeerServerImpl) {
        log.info("new peer join {}", peer)
    }

    override fun onDisconnect(peer: PeerImpl, server: PeerServerImpl) {
        log.info("peer disconnected {}", peer)
    }

    override fun onStop(server: PeerServerImpl) {}

    companion object {
        private val log = LoggerFactory.getLogger("net")
    }
}